src/rendering/RenderUtil.cc

```diff
@@ -1021,43 +868,7 @@ void RenderUtil::Update()
     }
   }
 
-  // update thermal camera
-  for (const auto &thermal : this->dataPtr->thermalCameraData)
```

Comment:
@iche033 - it looks like we used `this->dataPtr->thermalCameraData` instead of the [temporary `thermalCameraData` variable](https://github.com/ignitionrobotics/ign-gazebo/blob/334bca682f46b2139497b22f74adea11c4b37412/src/rendering/RenderUtil.cc#L578) when we originally did the thermal camera work in Dome. I'm guessing this was never an issue in the past because it's highly unlikely for the thermal camera data to update, but this should be fixed now in this refactor (you'll see that the `UpdateThermalCamera` helper method uses the temporary `thermalCameraData` variable).

Reply:
ah yes, good catch!